dbld/rules image-centos7 fails in python setup part

# syslog-ng

## Version of syslog-ng

Latest git snapshot

## Platform

Docker host: openSUSE 42.3, but using the CentOS 7 image

# Issue

## Failure

When trying to run ``dbld/rules image-centos7`` it fails in python setup part:
```
linux-6chp:~/syslog-ng # dbld/rules image-centos7
cd /root/syslog-ng/dbld/images/ && docker build --network=host -t balabit/syslog-ng-centos7 -f centos7.dockerfile .
Sending build context to Docker daemon    129kB
Step 1/21 : FROM centos:7
 ---> 49f7960eb7e4
Step 2/21 : ARG DISTRO=centos7
 ---> Using cache
 ---> 657157439c5f
Step 3/21 : LABEL maintainer "Andras Mitzki <andras.mitzki@balabit.com>, Laszlo Szemere <laszlo.szemere@balabit.com>"
 ---> Using cache
 ---> 87fcdc21f1a1
Step 4/21 : COPY helpers/* /helpers/
 ---> Using cache
 ---> 0cc24ff1e93f
Step 5/21 : RUN yum install -y   wget   epel-release
 ---> Using cache
 ---> e664067bd6ef
Step 6/21 : RUN yum install -y   python-pip   python-setuptools
 ---> Using cache
 ---> 0322c2d0a424
Step 7/21 : COPY required-pip/all.txt required-pip/${DISTRO}*.txt /required-pip/
 ---> Using cache
 ---> c82985d049f5
Step 8/21 : RUN cat /required-pip/* | grep -v '^$\|^#' | xargs pip install
 ---> Running in 67c5a4b07245
Collecting nose
  Downloading https://files.pythonhosted.org/packages/99/4f/13fb671119e65c4dce97c60e67d3fd9e6f7f809f2b307e2611f4701205cb/nose-1.3.7-py2-none-any.whl (154kB)
Collecting ply
  Downloading https://files.pythonhosted.org/packages/a3/58/35da89ee790598a0700ea49b2a66594140f44dec458c07e8e3d4979137fc/ply-3.11-py2.py3-none-any.whl (49kB)
Collecting pep8
  Downloading https://files.pythonhosted.org/packages/42/3f/669429ce58de2c22d8d2c542752e137ec4b9885fff398d3eceb1a7f5acb4/pep8-1.7.1-py2.py3-none-any.whl (41kB)
Collecting pylint
  Downloading https://files.pythonhosted.org/packages/9b/22/becc5a976c2b25c01f5d054b970a1da5803e61aa342d3729b6cc8d762f71/pylint-2.0.0.tar.gz (550kB)
    Complete output from command python setup.py egg_info:
    /usr/lib64/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'python_requires'
      warnings.warn(msg)
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-Ay9yTv/pylint/setup.py", line 177, in <module>
        install()
      File "/tmp/pip-build-Ay9yTv/pylint/setup.py", line 174, in install
        **kwargs)
      File "/usr/lib64/python2.7/distutils/core.py", line 112, in setup
        _setup_distribution = dist = klass(attrs)
      File "/usr/lib/python2.7/site-packages/setuptools/dist.py", line 265, in __init__
        self.fetch_build_eggs(attrs.pop('setup_requires'))
      File "/usr/lib/python2.7/site-packages/setuptools/dist.py", line 289, in fetch_build_eggs
        parse_requirements(requires), installer=self.fetch_build_egg
      File "/usr/lib/python2.7/site-packages/pkg_resources.py", line 618, in resolve
        dist = best[req.key] = env.best_match(req, self, installer)
      File "/usr/lib/python2.7/site-packages/pkg_resources.py", line 862, in best_match
        return self.obtain(req, installer) # try and download/install
      File "/usr/lib/python2.7/site-packages/pkg_resources.py", line 874, in obtain
        return installer(requirement)
      File "/usr/lib/python2.7/site-packages/setuptools/dist.py", line 339, in fetch_build_egg
        return cmd.easy_install(req)
      File "/usr/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 623, in easy_install
        return self.install_item(spec, dist.location, tmpdir, deps)
      File "/usr/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 653, in install_item
        dists = self.install_eggs(spec, download, tmpdir)
      File "/usr/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 849, in install_eggs
        return self.build_and_install(setup_script, setup_base)
      File "/usr/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 1130, in build_and_install
        self.run_setup(setup_script, setup_base, args)
      File "/usr/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 1115, in run_setup
        run_setup(setup_script, args)
      File "/usr/lib/python2.7/site-packages/setuptools/sandbox.py", line 69, in run_setup
        lambda: execfile(
      File "/usr/lib/python2.7/site-packages/setuptools/sandbox.py", line 120, in run
        return func()
      File "/usr/lib/python2.7/site-packages/setuptools/sandbox.py", line 71, in <lambda>
        {'__file__':setup_script, '__name__':'__main__'}
      File "setup.py", line 76, in <module>
    
      File "/usr/lib64/python2.7/distutils/core.py", line 112, in setup
        _setup_distribution = dist = klass(attrs)
      File "/usr/lib/python2.7/site-packages/setuptools/dist.py", line 269, in __init__
        _Distribution.__init__(self,attrs)
      File "/usr/lib64/python2.7/distutils/dist.py", line 287, in __init__
        self.finalize_options()
      File "/usr/lib/python2.7/site-packages/setuptools/dist.py", line 302, in finalize_options
        ep.load()(self, ep.name, value)
      File "build/bdist.linux-x86_64/egg/setuptools_scm/integration.py", line 10, in version_keyword
      File "build/bdist.linux-x86_64/egg/setuptools_scm/version.py", line 44, in _warn_if_setuptools_outdated
    setuptools_scm.version.SetuptoolsOutdatedWarning: your setuptools is too old (<12)
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-Ay9yTv/pylint/
You are using pip version 8.1.2, however version 10.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
The command '/bin/sh -c cat /required-pip/* | grep -v '^$\|^#' | xargs pip install' returned a non-zero code: 123
dbld/rules:107: recipe for target 'image-centos7' failed
make: *** [image-centos7] Error 123
```

--- a/dbld/images/centos7.dockerfile
+++ b/dbld/images/centos7.dockerfile
@@ -14,9 +14,11 @@ RUN yum install -y \
   python-pip \
   python-setuptools
 
+RUN pip install --upgrade pip
+
+
 COPY required-pip/all.txt required-pip/${DISTRO}*.txt /required-pip/
 RUN cat /required-pip/* | grep -v '^$\|^#' | xargs pip install
-
 
 COPY required-yum/all.txt required-yum/${DISTRO}*.txt /required-yum/
 RUN cat /required-yum/* | grep -v '^$\|^#' | xargs yum install -y

